### PR TITLE
add txt files with attribute and reach names

### LINKED ISSUE #227: Add text files for attributes and segments to use in different model configurations
For integration with river-dl, it will be more convenient to create txt files with the attribute names corresponding to each of the combinations:
- min static + dynamic
- static + dynamic
- dynamic

And for the spatial train/test split, also a txt file with the test set reach IDs.

These can be supplied to function args for RGCN model training

### PR DESCRIPTION
**Pull request by jds485**
Creates 1-column text files with: 
- 3 sets of attributes (dynamic, min static + dynamic, static + dynamic)
- the reaches in the spatial test set, and within validation sets used during training.
- the start and end dates within training, validation, and testing

These files are loaded into snakemake workflows for the RGCN model training.

Creates 4-column text files with:
- PRMS_segid
- Date
- obs (observations)
- pred (predictions)

These files are loaded into a snakemake workflow to compute performance metrics for the RF models.

Also adds output directories for RGCN model runs

Closes #227

### CODE REVIEW

**Comment by jds485** [comment]
Thanks, Margaux!

> Couldn't run with Docker but will aim to address this later this week if I find the time.

What was the issue with using the container?

> ...upload the latest versions of these to Sharepoint as csvs, particularly the different dynamic/static attributes txt files so that they can be defined there in a collaborative way.

Are csv files better for the model archive? I can make a target that writes csv files. Would it be better to have 1 csv file containing all of these attributes (i.e., unique() across the 3 sets of attributes)? I am not planning to change these attribute files, so I can upload to SharePoint after this PR is merged. Automatic upload would be neat, though. Do you know how to set that up? I don't think it's necessary because these files are not likely to change, but I'm interested in learning how to do that! Maybe with a Globus command to send from Caldera to local OneDrive?

Define = describe what the attribute represents? Most of the static attributes (CAT and TOT) have descriptions in the Wieczorek data release. Some dynamic as well (land cover).

File: 4_predict.R [hunk 1]
```diff
@@ -569,5 +570,102 @@ p4_targets_list <- list(
...
+  ),
+  tar_target(p4_static_dynamic_attrs_txt,
+             write_attrs_files(p4_selected_static_dynamic_attrs$input_data$testing,
+                                  drop_cols = c("PRMS_segid", "Date", "mean_value", "data_type"),
+                                  filepath = "4_predict/out/static_dynamic_attrs.txt"),
+             deployment = 'main',
+             format = 'file',
+             repository = 'local'
+  ),
+  tar_target(p4_spatial_cv_reaches_txt,
```
[the remark this answers is not in the document]
Comment (jds485) [reply 4_predict.R:602]: See most recent commit. I moved comments around to describe groups of targets

**Comment by jds485** [comment]
@msleckman: I think I addressed your comments. I had a question for you about adding an additional csv target. Let me know what you'd prefer.

**Comment by jds485** [comment]
Okay, thanks! I'll make a csv target specifically for metadata then.

> I have to think through how Wiezoreck's definitions can be used here. I'll have inquire with others.

Check with Andrew Sekellick when you're ready to start doing this. He's working on a data release for the FHWA project and is planning to use the metadata from the Wieczorek SB entries.

**Comment by lekoenig** [comment]
> Check with Andrew Sekellick when you're ready to start doing this. He's working on a data release for the FHWA project and is planning to use the metadata from the Wieczorek SB entries.

Great idea to check 🤩 @msleckman this might also be useful for the PGDL-GW release you're working on. Let Jared or I know if you want us to connect you with Andrew.

**Comment by jds485** [comment]
@msleckman: I added the csv attribute file target. Let me know if this is good to merge
